src/app/clusters/bindings/BindingManager.h

```diff
@@ -0,0 +1,173 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/CASESessionManager.h>
+#include <app/server/Server.h>
+#include <app/util/binding-table.h>
+
+namespace chip {
+
+/**
+ * Application callback function when a cluster associated with a bound peer changes.
+ *
+ * The connection is managed by the stack and peer_device is guaranteed to be available.
+ * The application shall decide the content to be sent to the peer.
+ *
+ * E.g. The application will send on/off commands to peer for the OnOff cluster.
+ *
+ */
+using BoundDeviceChangedHandler = void (*)(EndpointId localEndpoint, EndpointId remoteEndpoint, ClusterId clusterId,
+                                           DeviceProxy * peer_device, Optional<GroupId> group);
```

Comment:
From the current implementation of `AssociateWithGroup` https://github.com/project-chip/connectedhomeip/blob/master/src/controller/CHIPCluster.cpp#L45, we still need a `peer_device` to talk to a group. I have to say I don't know where the `DeviceProxy` object will come from when talking to a group.

Reply:
The callee in this case is not even using CHIPCluster.  All it needs to talk to a group, given the API it's using, is a group session and an exchange manager.  And it can get a group session given an exchange manager and a group id.

The only thing the `DeviceProxy` is used for in `AssociateWithGroup` is getting the exchange manager.

In this case, though, the callee knows exactly where to get an exchange manager: from `chip::Server::GetInstance()`, which is where in the end we're getting the CASESessionManager from.  So all the callee needs to get to do a group invoke is the local endpoint, cluster id, group id.